azurerm/internal/services/springcloud/spring_cloud_java_deployment_resource_test.go

```diff
@@ -0,0 +1,174 @@
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccSpringCloudJavaDeployment_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_java_deployment", "test")
+	r := SpringCloudJavaDeploymentResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccSpringCloudJavaDeployment_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_java_deployment", "test")
+	r := SpringCloudJavaDeploymentResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r SpringCloudJavaDeploymentResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := parse.SpringCloudDeploymentID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.AppPlatform.DeploymentsClient.Get(ctx, id.ResourceGroup, id.SpringName, id.AppName, id.DeploymentName)
+	if err != nil {
+		return nil, fmt.Errorf("reading Spring Cloud Deployment %q (Spring Cloud Service %q / App %q / Resource Group %q): %+v", id.DeploymentName, id.SpringName, id.AppName, id.ResourceGroup, err)
+	}
+
+	return utils.Bool(resp.Properties != nil), nil
+}
+
+func (r SpringCloudJavaDeploymentResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_spring_cloud_java_deployment" "test" {
+  name                = "acctest-scjd%s"
+  spring_cloud_app_id = azurerm_spring_cloud_app.test.id
+}
+`, r.template(data), data.RandomString)
+}
+
+func (r SpringCloudJavaDeploymentResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_spring_cloud_java_deployment" "import" {
+  name                = azurerm_spring_cloud_java_deployment.test.name
+  spring_cloud_app_id = azurerm_spring_cloud_java_deployment.test.spring_cloud_app_id
+}
+`, r.basic(data))
+}
+
+func (r SpringCloudJavaDeploymentResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_spring_cloud_java_deployment" "test" {
+  name                = "acctest-scjd%s"
+  spring_cloud_app_id = azurerm_spring_cloud_app.test.id
+  cpu                 = 2
+  memory_in_gb        = 4
+  instance_count      = 2
+  jvm_options         = "-XX:+PrintGC"
+  runtime_version     = "Java_8"
```

Comment:
Missed this one last pass, can we have this as `Java_11` to it's a changed value in the `update` test?